classes/package/package_info.php

```diff
@@ -0,0 +1,179 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_questionpy\package;
+
+defined('MOODLE_INTERNAL') || die;
+
+use moodle_exception;
+use qtype_questionpy\array_converter\array_converter;
+use qtype_questionpy\array_converter\converter_config;
+
+/**
+ * Represents an overview of available QuestionPy packages on the application server.
+ *
+ * @package    qtype_questionpy
+ * @copyright  2024 Jan Britz, TU Berlin, innoCampus - www.questionpy.org
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class package_info extends package_base {
+    /**
+     * Checks whether this package exists or not.
+     *
+     * @return bool|int either false or the DB id
+     * @throws moodle_exception
+     */
+    public function exists(): bool|int {
+        global $DB;
+        return $DB->get_field('qtype_questionpy_package', 'id', [
+            'shortname' => $this->shortname,
+            'namespace' => $this->namespace,
+        ]);
+    }
```

Comment:
Evtl. aussagekräftiger:
```suggestion
    /**
     * Gets the row id if this package is already stored in the DB, or false otherwise.
     *
     * @return false|int either false or the DB id
     * @throws moodle_exception
     */
    public function get_db_id(): false|int {
        global $DB;
        return $DB->get_field('qtype_questionpy_package', 'id', [
            'shortname' => $this->shortname,
            'namespace' => $this->namespace,
        ]);
    }
```